pkg/apis/vsphere/v1/template_genclient.go

```diff
@@ -0,0 +1,101 @@
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/url"
+	"path"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+)
+
+var (
+	// ErrTemplateOperationNotSupported is returned if operations other than Get and List are performed
+	ErrTemplateOperationNotSupported = fmt.Errorf("%w: Template only supports Get and List operations", api.ErrOperationNotSupported)
+)
+
+// EndpointURL returns the URL where to retrieve objects of type Template (only Get and List operations supported)
+func (t *Template) EndpointURL(ctx context.Context) (*url.URL, error) {
+	_, err := t.operationFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return url.ParseRequestURI(
+		fmt.Sprintf("/api/vsphere/v1/provisioning/templates.json/%s/%s", t.Location.Identifier, t.Type),
+	)
+}
+
+// HasPagination disables pagination for Template API (not supported by engine)
+func (t *Template) HasPagination(ctx context.Context) (bool, error) {
+	return false, nil
+}
+
+// FilterRequestURL removes the Identifier from URL on Get operations (template needs to be parsed from list response)
+func (t *Template) FilterRequestURL(ctx context.Context, url *url.URL) (*url.URL, error) {
+	op, err := t.operationFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if op == types.OperationGet {
+		url.Path = path.Dir(url.Path)
+	}
+
+	q := url.Query()
+	q.Set("page", "1")
+	q.Set("limit", "1000")
```

Comment:
is the query on the URL actually updated with this? I seem to remember it's not and you have to do something like this:
```go
url.RawQuery = q.Encde()
```

Reply:
did you check if this is needed?

Reply:
Yes, it's needed. `url.String()` did not contain the query params without.